src/track/keyutils.cpp

```diff
@@ -698,3 +698,60 @@ int KeyUtils::keyToCircleOfFifthsOrder(mixxx::track::io::key::ChromaticKey key,
         return s_sortKeysCircleOfFifthsLancelot[static_cast<int>(key)];
     }
 }
+
+// returns the pitch difference between two tracks, when they are played at the
+// same tempo. Assumes all inputs are valid.
+double KeyUtils::trackSyncPitchDifference(
+        mixxx::track::io::key::ChromaticKey key1,
+        mixxx::Bpm bpm1,
+        mixxx::track::io::key::ChromaticKey key2,
+        mixxx::Bpm bpm2) {
+    // convert minor keys to their relative majors
+    // we are interested in the notes used, not the mode
+    key1 = minorToRelativeMajor(key1);
+    key2 = minorToRelativeMajor(key2);
+
+    // calculate the pitch delta for each track when stretched to a tempo of 100BPM
+    const double delta1 = powerOf2ToSemitoneChange(100 / bpm1.value());
+    const double delta2 = powerOf2ToSemitoneChange(100 / bpm2.value());
+
+    // get the resulting key for each track at 100BPM
+    const double resPitch1 = normalizePitch(keyToNumericValue(key1) - 1 + delta1);
+    const double resPitch2 = normalizePitch(keyToNumericValue(key2) - 1 + delta2);
+
+    // return the pitch difference when both tracks are played at the same tempo
+    return normalizePitch(resPitch1 - resPitch2);
```

Comment:
We have a pitch C and C' which both are C finally. Does it make sens to introduce here a also a circle (actually a spirale) and introduce a function taht calculates the shortes pitch distance up or down. This is probably effectively the sam we have but with a nicer API.  

This may than also consider the sync feature which uses 0.5 and 2.0 if this is the shorter pitch shift:  
https://github.com/mixxxdj/mixxx/blob/42ef112d0a8f70dd256acd7da2b9950084f9ac07/src/engine/sync/synccontrol.cpp#L16-L18

Reply:
Yes!